api/envoy/config/filter/http/rbac/v2/rbac.proto

```diff
@@ -12,18 +12,16 @@ import "gogoproto/gogo.proto";
 // Role-Based Access Control :ref:`configuration overview <config_http_filters_rbac>`.
 
 message RBAC {
-  // Specify the RBAC rules to be applied globally
-  config.rbac.v2alpha.RBAC rules = 1 [(validate.rules).message.required = true];
+  // Specify the enforcement mode RBAC rules to be applied globally.
+  // Disable the filter for enforcement mode if rules isn't set.
+  config.rbac.v2alpha.RBAC rules = 1;
```

Comment:
IMO, the use of "enforcement" and "permissive" modes come off as jargon (and it's mainly an implementation detail of the engine) when the behavior is the permissive rules are evaluated but do not affect the actual rules. Is there better language we can use that explains this? Is there existing behavior in Envoy that we can match the terminology on (@mattklein123)?

Reply:
I would probably do "Enforcing" and "Shadow" or something like that, but open to other ideas.

Reply:
renamed permissive -> shadow, here and other places in this PR.